src/core/ext/filters/client_channel/xds/xds_client.cc

```diff
@@ -812,12 +812,17 @@ void XdsClient::ChannelState::AdsCallState::SendMessageLocked(
         GRPC_ERROR_REF(state.error), !sent_initial_message_);
     state.subscribed_resources[xds_client()->server_name_]->Start(Ref());
   } else if (type_url == XdsApi::kRdsTypeUrl) {
-    resource_names.insert(xds_client()->lds_result_->route_config_name);
+    std::string name = (xds_client()->lds_result_.has_value()
+                            ? xds_client()->lds_result_->route_config_name
+                            : xds_client()->server_name_);
+    resource_names.insert(name);
     request_payload_slice = xds_client()->api_.CreateRdsRequest(
-        xds_client()->lds_result_->route_config_name, state.version,
-        state.nonce, GRPC_ERROR_REF(state.error), !sent_initial_message_);
-    state.subscribed_resources[xds_client()->lds_result_->route_config_name]
-        ->Start(Ref());
+        name, state.version, state.nonce, GRPC_ERROR_REF(state.error),
+        !sent_initial_message_);
+    auto resource_it = state.subscribed_resources.find(name);
+    if (resource_it != state.subscribed_resources.end()) {
```

Comment:
This also looks wrong to me.  If we don't actually have an entry in the subscribed resources list, why are we sending a request in the first place?

Actually, I wonder if a better approach would be to construct the list of resource names from the list of subscribed resources, just like we do for CDS and EDS requests.  But we would need to be careful about the case where the list is empty -- if that happens on the very first request for a given resource type, we should not send the request at all, since that would be interpretted as a wildcard request.